invenio_rdm_records/resources/resources.py

```diff
@@ -227,3 +241,167 @@ def search(self):
+    @response_handler()
+    def base(self):
+        """Base."""
+        item = self.service.get_file(
+                identity=g.identity,
+                uuid=resource_requestctx.view_args["uuid"],
+            )
+        raise RedirectException(item["links"]["iiif_info"])
+
+    @cross_origin(origin="*", methods=["GET"])
+    @with_iiif_content_negotiation(IIIFInfoV2JSONSerializer)
+    @iiif_request_view_args
+    @response_handler()
+    def info(self):
+        """Get IIIF image info."""
+        item = self.service.get_file(
+                identity=g.identity,
+                uuid=resource_requestctx.view_args["uuid"],
+            )
+        return item.to_dict(), 200
+
+    @cross_origin(origin="*", methods=["GET"])
+    @iiif_request_view_args
+    def image_api(self):
+        """IIIF API Implementation.
+
+        .. note::
+            * IIF IMAGE API v1.0
+                * For more infos please visit <http://iiif.io/api/image/>.
+            * IIIF Image API v2.0
+                * For more infos please visit <http://iiif.io/api/image/2.0/>.
+            * The API works only for GET requests
+            * The image process must follow strictly the following workflow:
+                * Region
+                * Size
+                * Rotation
+                * Quality
+                * Format
+        """
+        image_format = resource_requestctx.view_args["image_format"]
+        uuid = resource_requestctx.view_args["uuid"]
+        region = resource_requestctx.view_args["region"]
+        size = resource_requestctx.view_args["size"]
+        rotation = resource_requestctx.view_args["rotation"]
+        quality = resource_requestctx.view_args["quality"]
+        to_serve = self.service.image_api(
+            identity=g.identity,
+            uuid=uuid,
+            region=region,
+            size=size,
+            rotation=rotation,
+            quality=quality,
+            image_format=image_format,
+        )
+        # decide the mime_type from the requested image_format
+        mimetype = self.config.supported_formats.get(
+            image_format, "image/jpeg"
+        )
+        # TODO: get from cache on the service image.last_modified
+        last_modified = None
+        send_file_kwargs = {"mimetype": mimetype}
+        # last_modified is not supported before flask 0.12
+        if last_modified:
+            send_file_kwargs.update(last_modified=last_modified)
+
+        if "dl" in request.args:
+            filename = secure_filename(request.args.get("dl", ""))
+            if filename.lower() in {"", "1", "true"}:
+                filename = "{0}-{1}-{2}-{3}-{4}.{5}".format(
+                    uuid, region, size, quality, rotation, image_format
+                )
+            send_file_kwargs.update(
+                as_attachment=True,
+                attachment_filename=secure_filename(filename),
+            )
+        if_modified_since_raw = request.headers.get("If-Modified-Since")
```

Comment:
Minor: The header should be parsed and added in the ``resource_requestctx.headers``.